### PR TITLE
Add subpath redirects

### PR DESCRIPTION
**Pull request by huynhtrankhanh**

### CODE REVIEW

**Comment by Zimmi48** [comment]
Thanks for your contribution!
For the record, I do not think that the CI in this repo is of any use for pull requests. However, you can test your changes locally using `make && make run`. If I'm not mistaken, you only need OCaml as a dependency for that.

File: pages/404.html [hunk 1]
```diff
@@ -0,0 +1,14 @@
+<#include "incl/macros.html">
+<script>
+  const path = window.location.pathname;
+  const stdlibRegex = /^\/(stdlib|library)\//;
+  const refmanRegex = /^\/distrib\/(.*)\/refman\//;
+  if (stdlibRegex.test(path)) {
+    window.location = path.replace(stdlibRegex, "/doc/<#CURRENTVERSIONTAG>/stdlib/");
+  } else if (refmanRegex.test(path)) {
+    const version = path.match(refmanRegex)[1];
+    window.location = path.replace(refmanRegex, "/doc/" + version + "/refman/");
```
Comment (Zimmi48) [inline pages/404.html:10]: The case of `distrib/current/refman` will need to be handled specially (using the `#CURRENTVERSIONTAG`). BTW, this case also exists for the stdlib doc (as in `/distrib/current/stdlib` or `/distrib/<version>/stdlib`).

**Comment by Zimmi48** [comment]
> Thanks for your contribution! For the record, I do not think that the CI in this repo is of any use for pull requests. However, you can test your changes locally using `make && make run`. If I'm not mistaken, you only need OCaml as a dependency for that.

Actually, it's unclear that this method of testing works because the simple server that it runs doesn't seem to pick up the 404.html page (whereas GitHub Pages possibly could).

**Comment by huynhtrankhanh** [comment]
I tested this on GitHub Pages. https://github.com/htk-coq-website-fix/htk-coq-website-fix.github.io/tree/master

There might be missing cases, of course. I don't know enough about all the old links.

**Comment by huynhtrankhanh** [comment]
It's night where I live. It might be best to merge the PR now and then add the missing cases later. I believe I handled the existing cases correctly, and you can independently test this on the fork (https://htk-coq-website-fix.github.io)

File: pages/404.html (hunk 1 above)
Comment (huynhtrankhanh) [reply pages/404.html:10]: The distrib/current/refman is already handled by meta refresh redirect (https://htk-coq-website-fix.github.io/distrib/current/refman)

Comment (huynhtrankhanh) [reply pages/404.html:11]: The same goes for /distrib/current/stdlib

**Comment by huynhtrankhanh** [comment]
I believe I handled all cases

**Comment by Zimmi48** [comment]
Thanks a lot! I don't want to make you stay up late. It's also already pretty late here. I'll do some further testing using your website and merge in the morning. If there are additional cases to handle (there probably are), they can be added later on based on the model you provided.

File: pages/404.html (hunk 1 above)
Comment (Zimmi48) [reply pages/404.html:11]: This actually does not handle sub-paths. Here is an example of broken URL: https://htk-coq-website-fix.github.io/doc/current/refman/addendum/type-classes.html (already broken before this PR of course).

**Pull request merged**